backend/src/appointment/database/repo/schedule.py

```diff
@@ -87,6 +87,27 @@ def has_slot(db: Session, schedule_id: int, slot_id: int):
     return db_slot and db_slot.schedule_id == schedule_id
 
 
+def has_valid_availability(schedule: schemas.ScheduleValidationIn):
+    """check if availability is valid"""
+    size = len(schedule.availabilities)
+    if size > 0:
+        # Make sure, that the availabilities are sorted by weekday AND by start time. This is important for checking
+        # validity of times of adjacent availabilities at the same day
+        for i, a in enumerate(sorted(schedule.availabilities, key = lambda x: (x.day_of_week.value, x.start_time))):
+            # Check valid times and duration of a single availability entry
+            if not utils.is_valid_time_range(a.start_time, a.end_time, schedule.slot_duration):
+                return False
+            # Chack valid predecessor
+            if (i > 0 and a.day_of_week == schedule.availabilities[i-1].day_of_week
+                and not utils.is_valid_time_range(schedule.availabilities[i-1].end_time, a.start_time)):
+                return False
+            # Check valid successor
+            if (i < size-1 and a.day_of_week == schedule.availabilities[i+1].day_of_week
+                and not utils.is_valid_time_range(a.end_time, schedule.availabilities[i+1].start_time)):
+                return False
```

Comment:
Let me know if you think this is redundant. It felt right to check both sides when implementing this 😇

Reply:
Ok thanks for clarifying, no I think renaming to `all_availability_is_valid` would be good

Reply:
Alright, renamed now!